qos/jsonrpc/request.go

```diff
@@ -23,6 +23,32 @@ type Request struct {
 	Params  Params  `json:"params,omitempty"`
 }
 
+// NewRequest constructs a new JSONRPC request from the given ID, method, and optional parameters.
+//
+// For example:
+// id - 1
+// method - eth_getBalance
+// params - ["0x28C6c06298d514Db089934071355E5743bf21d60", "0xe71e1d"]
+//
+// The above request would be serialized as:
+// {"jsonrpc":"2.0","id":1,"method":"eth_getBalance","params":["0x28C6c06298d514Db089934071355E5743bf21d60", "0xe71e1d"]}
+func NewRequest(id int, method Method, params ...any) Request {
```

Comment:
Please reconsider this (specifically use of `any`):
- All fields of `Request` are already public: `NewRequest` does not add any functionality and is not GO idiomatic.
- A helper method to build `Params` from a typed set of values, e.g. `BuildParamsFromStrings([]string)`, would be much easier to read/maintain.

Reply:
OK, I've updated to remove `NewRequest` and instead create a ` jsonrpc.BuildArrayParamsFromStrings` helper method.

Please see commit: 
https://github.com/buildwithgrove/path/pull/194/commits/cf70274f346fc5a97b0036219293e382661523cb